tests/csapi/keychanges_test.go

```diff
@@ -0,0 +1,111 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestKeyChangesLocal(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	password := "$uperSecretPassword"
+	bob := deployment.RegisterUser(t, "hs1", "bob", password, false)
+	unauthedClient := deployment.Client(t, "hs1", "")
+
+	t.Run("Parallel", func(t *testing.T) {
+		t.Run("New login should create a device_lists.changed entry", func(t *testing.T) {
+			mustUploadKeys(t, bob)
+
+			roomID := alice.CreateRoom(t, map[string]interface{}{"preset": "private_chat"})
+			alice.InviteRoom(t, roomID, bob.UserID)
```

Comment:
Should also work with a public chat so you can skip the invite? Would be fractionally faster.